server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -172,8 +174,11 @@ public String executor() {
                 new ExtensionRestRequest(method, path, params, contentType, content, requestIssuerIdentity),
                 restExecuteOnExtensionResponseHandler
             );
+            inProgressFuture.orTimeout(5, TimeUnit.SECONDS);
             try {
-                inProgressLatch.await(5, TimeUnit.SECONDS);
+                if (inProgressFuture.isCompletedExceptionally()) {
+                    inProgressFuture.get();
```

Comment:
The `InterruptedException` error message should be for a `TimeoutException`.  

May want another one to handle interrupted or canceled, and actually return a failure on `ExecutionException`.